test/spmd/test_dynamo_spmd.py

```diff
@@ -53,6 +59,22 @@ def test_dynamo_spmd_basic(self):
     # TODO(JackCaoG): add counter checks after ExecuteReplicated also creates
     # a ExecuteMetric.
 
+  def test_dynamo_spmd_basic_with_custom_mark_sharding_op(self):
```

Comment:
nit, maybe rename it something similar to `test_dynamo_spmd_mark_sharding_outside_of_compile`